src/integrationTest/groovy/wooga/gradle/unity/tasks/AddUPMPackagesTaskIntegrationSpec.groovy

```diff
@@ -0,0 +1,34 @@
+package wooga.gradle.unity.tasks
+
+import com.wooga.spock.extensions.unity.UnityPathResolution
+import com.wooga.spock.extensions.unity.UnityPluginTestOptions
+import com.wooga.spock.extensions.uvm.UnityInstallation
+import net.wooga.uvm.Installation
+import spock.lang.Requires
+import wooga.gradle.unity.UnityTaskIntegrationSpec
+
+class AddUPMPackagesTaskIntegrationSpec extends UnityTaskIntegrationSpec<AddUPMPackages> {
+
+    @Requires({ os.macOs })
+    @UnityPluginTestOptions(unityPath = UnityPathResolution.Default)
+    @UnityInstallation(version = "2019.4.24f1", cleanup = false)
+    def "adds Packages"(Installation unity) {
+        given: "an unity3D project"
+        def project_path = "build/test_project"
+        environmentVariables.set("UNITY_PATH", unity.getExecutable().getPath())
+        appendToSubjectTask("""
+                createProject = "${project_path}"
+                buildTarget = "Android"
+                manifestPath.set("${project_path}/Packages/manifest.json")
+                upmPackages.put("com.unity.testtools.codecoverage", "1.1.0")
+            """.stripIndent())
+
```

Comment:
It would be nice to assert that then manifest file does not exists before the "when" clause.

Reply:
But you can.
just move the line `def manifestFile = new File(new File(projectDir, projectPath), "Packages/manifest.json")` up and run `assert !manifestFile.exists()`